solr/app.py

```diff
@@ -17,10 +19,37 @@ def create_app():
     app.url_map.strict_slashes = False
     Consul(app)  # load_config expects consul to be registered
     load_config(app)
+    if config:
+        app.config.update(config)
+        
+    db.init_app(app)
     logging.config.dictConfig(
         app.config['SOLR_SERVICE_LOGGING']
     )
     
+    ## pysqlite driver breaks transactions, we have to apply some hacks as per
```

Comment:
this code here serves a different purpose, it fixes sqlite problems with transactions. Sqlite driver is confused about when to call 'BEGIN' so here we work around it

as for the problem of testing a different engine, os.environ.get('SQLALCHEMY_URL', 'sqlite:///') would do the job perfectly - i don't think the impl specific test is the right way - we don't have postgres as a pre-requisite. ```sql_session.execute('create table users (id integer, random integer);')```